mycore-restapi/src/main/java/org/mycore/restapi/v2/annotation/MCRRestRequiredPermission.java

```diff
@@ -38,5 +41,13 @@
      *
      * @return the permission value
      */
-    String value();
+    String value() default "";
+
+    /**
+     * Resolves the required permission.
+     *
+     * @return the permission value
+     */
+    Class<? extends MCRRestRequiredPermissionResolver> resolver()
+        default MCRRestRequiredPermissionDefaultResolver.class;
```

Comment:
This annotation is on a given endpoint, how does it have to resolve a permission (a.k.a. privilege) from the request? Can you elaborate a bit more on the use case?
For example if I write on a PUT endpoint @MCRRequiredPermission("myWritePerm") that's one permission I would want there. Where comes additional usage from the dynamic part in here?

Reply:
One case would be if the permission has to be derived from the properties of the resource itself. 
Assume there are read and write keys that are to be managed via one endpoint. For updating, different permissions are required.  If I want to update an access key, I do not know directly whether it is a read or write key and must first determine the corresponding permission.

Reply:
I see. Than I would suggest a quite different approach. A resolver should decide if the request is allowed or not. It should get the ResourceInfo and the ContainerRequestContext object and than make a final decision. A default implementation could than implement the current behavior of using the HTTP method and path parameters. This should come with a new Annotation then.

Reply:
I had already thought about this.  However, it should always be ensured that the basic check resp. `#checkRestAPIAccess` is executed, which is why I opted for this approach. I had also considered refactoring `#checkRestAPIAccess` and then providing an abstract resolver, but I found the current approach  easier to use.